cilib/service/snap.py

```diff
@@ -157,6 +174,30 @@ def sync_all_track_snaps(self):
                     self.log(f"Found no stable branches ({_version}), skipping.")
                     continue
 
+            # Go versions within a single track can update over time
+            # just because 1.28.0 builds with go/1.20, 1.28.13 may use go/1.22
+            # so we need to confirm this branch is using the correct go version
+            # and trigger new builds if it changes.
+            branch = f"v{latest_branch_version}"
+            branch_ver = semver.VersionInfo.parse(latest_branch_version)
+            if content := self.snap_model.base.cat(branch, "/snapcraft.yaml"):
+                go_version = self.go_version(branch_ver)
+                if go_version not in content:
+                    self.log(
+                        f"Go version mismatch for {branch}, updating snapcraft.yaml"
+                    )
+                    with tempfile.TemporaryDirectory() as tmpdir:
+                        branch = f"v{latest_branch_version}"
+                        src_path = Path(tmpdir) / self.snap_model.src
+                        self.snap_model.base.clone(cwd=tmpdir)
+                        self.snap_model.base.checkout(branch, cwd=str(src_path))
+                        self.template_snapcraft_yml(
+                            src_path, branch, branch_ver, commit_msg="Update Snapcraft"
+                        )
+                    self.log("Go version changed to {go_version}, building new snap")
+                    self._create_recipe(_version, branch)
+                    continue
```

Comment:
If the go version changes, push the change to launchpad and run the recipe to rebuild the snap.